### PR TITLE
use .env file to provide external IP address

### PR DESCRIPTION
**Pull request by belegnar**
## What type of PR?

enhancement

## What does this PR do?

prettyfies docker-compose.yml been generated

## Prerequistes
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

### CODE REVIEW

**Comment by Nebukadneza** [comment]
I know that `.env` is somewhat the standard, and evaluated by default, but since a while the the `docker-compose.yml` offered by the setup utility sets the env-file to `mailu.env`. I agree that pulling out the IP to a single place is quite nice, but why should we add a third file, `.env` instead of using the configuration-bearing `mailu.env` which already exists? Would you also be happy with putting it in `mailu.env`?

Thanks for your addition and cleanups though ^_^.

**Comment by belegnar** [comment]
Because `mailu.env` is not available _before_ build, so you can't reference vars from it in `docker-compose.yaml`
https://vsupalov.com/docker-arg-env-variable-guide/

**Comment by muhlemmer** [comment]
> Because `mailu.env` is not available _before_ build, so you can't reference vars from it in `docker-compose.yaml`
> https://vsupalov.com/docker-arg-env-variable-guide/

Variables in setup are appointed in the python app. Why the additional `.env`? How is the additional env parsed from inside the app?

File: core/postfix/start.py [hunk 1]
```diff
@@ -36,7 +36,9 @@ def start_podop():
 
 if os.path.exists("/overrides/postfix.cf"):
     for line in open("/overrides/postfix.cf").read().strip().split("\n"):
-        os.system('postconf -e "{}"'.format(line))
+        line = line.strip()
+        if line and line[0] != "#":
+            os.system('postconf -e "{}"'.format(line))
```
Comment (muhlemmer) [inline core/postfix/start.py]: What is the puprose of this change? it is not described in the PR title or body? This should go in a separate PR.

File: docs/faq.rst [hunk 2]
```diff
@@ -242,6 +242,9 @@ correct syntax. The following file names will be taken as override configuration
 
 *Issue reference:* `206`_.
 
+ keep in mind, that commands from ``postfix.cf`` are injected via system call to ``postconf -e``
+ so symbols been used should be escaped
+
```
Comment (muhlemmer) [inline docs/faq.rst]: I'm very OK with including this into the FAQ, but has nothing to do with the topic of this PR? Perhaps belongs in a PR with the changes in `core/postfix/start.py`?

File: setup/flavors/compose/.env [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{% if bind4 %}
+EXTERNAL_IP={{ bind4 }}
+{% else %}
+EXTERNAL_IP=0.0.0.0
+{% endif %}
```
Comment (muhlemmer) [inline setup/flavors/compose/.env]: We don't [officially support](https://mailu.io/1.7/compose/setup.html#bind-address) running on `0.0.0.0`. As such, I don't think we should provide this as a default:

> If the address is not configured directly (NAT) on any of the network interfaces or if you would simply like the server to listen on all interfaces, use 0.0.0.0 and ::. Note that running is this mode is not supported and can lead to [issues](https://github.com/Mailu/Mailu/issues/641).

**Comment by kaiyou** [comment]
Sorry about my current opinion, but really, if the only gain is to be able to parametrize the IP address, it is probably not worth it multiplying the number of config files the admin has to manage.

We could discuss going back to `.env` in a separate issue/discussion.

**Comment by micw** [comment]
I reviewed it and came to the same conclusion: We should close this in favor of separate PRs with a clean scope.

**Comment by muhlemmer** [comment]
Closing as discussed [here](https://matrix.to/#/!RJFCFtixHgPhzacdhW:tedomum.net/$1570831252137459GIvHn:matrix.org?via=tedomum.net&via=matrix.org&via=ghostdub.de)
